modules/community-algorithms/pages/k-means-clustering.adoc

```diff
@@ -0,0 +1,83 @@
+= k-Means Clustering
+
+.Supported Graph Characteristics
+****
+[cols='1']
+|===
+^|Unweighted edges
+^|Weighted edges
+^|Directed edges
+^|Undirected edges
+^|Homogeneous vertex types
+|===
+
+Algorithm link: link:https://github.com/tigergraph/gsql-graph-algorithms/tree/master/algorithms/Community/k_means[k-Means Clustering]
+
+****
+
+In the k-Means Clustering algorithm, vertices are gradually sorted into an arbitrary number _k_ of clusters based on averaging their properties.
+
+First, _k_ vertices are selected at random to act as centroids, or the center of their clusters. The first clusters are defined as consisting of the vertices closest to those centroids. The borders of these clusters are refined iteratively by calculating the mean position of all points in the clusters, then by finding the closest vertices to those mean position points to act as the new centroids.
+
+The last two steps are repeated many more times until each iteration no longer changes the location of the centroids or the vertices belonging to the clusters.
+
+Vertices in the graph need to have an attribute in their schemas that is a list of doubles or floats (an embedding) for this algorithm to work. This is because the distances are calculated based on the vertex embeddings rather than any other factor.
+
+In some use cases, an embeddings algorithm like FastRP can be used to create these embeddings.
```

Comment:
If FastRP is published, we should link to it. Otherwise, maybe we can link to a Wikipedia link